common/djangoapps/third_party_auth/models.py

```diff
@@ -784,16 +784,23 @@ def get_config(self):
             conf['attr_defaults'][field] = default
 
         # Now get the data fetched automatically from the metadata.xml:
-        data = SAMLProviderData.current(self.entity_id)
-        if not data or not data.is_valid():
+        data_records = SAMLProviderData.objects.filter(entity_id=self.entity_id)
```

Comment:
I agree with @johnnagro -- you can add a utility method encapsulating the filter and validity check and DRY this code, returning only the valid records by default + optionally returning both valid and invalid records.